### PR TITLE
Checks for Response Statements for Implemented Requires for Controls

### LINKED ISSUE #21: Every Statement in a profile with a Response Type should have a corresponding response (of length > 20)

### PR DESCRIPTION
**Pull request by ohsh6o**
Closes #21.

### CODE REVIEW

**Comment by ohsh6o** [comment]
I guess I cannot formally include you in the review list for now, @wendellpiez.

**Review by mike-stern — COMMENTED** [review]
Good job creating positive and negative test scenarios.

**Review by mike-stern — APPROVED** [review]

File: resources/validations/src/ssp.sch [hunk 1]
```diff
@@ -236,5 +236,49 @@
...
+    <sch:rule context="/o:system-security-plan/o:control-implementation/o:implemented-requirement/o:statement/o:by-component">
+        <sch:let name="component-ref" value="./@component-uuid"/>
+        <sch:assert role="warning" id="invalid-component-match" test="/o:system-security-plan/o:system-implementation/o:component[@uuid = $component-ref] => exists()"
+            >Response statment <sch:value-of select="../@statement-id"/> with component reference UUID '<sch:value-of select="$component-ref"/>' is not in the system implementation inventory, and cannot be used to define a control.</sch:assert>
+        <sch:assert role="error" id="missing-component-description" test="./o:description => exists()"
+            >Response statement <sch:value-of select="../@statement-id"/> has a component, but that component is missing a required description node.</sch:assert>"
+    </sch:rule>
+
+    <sch:rule context="/o:system-security-plan/o:control-implementation/o:implemented-requirement/o:statement/o:by-component/o:description">
+        <sch:let name="required-length" value="20"/>
```
[the remark this answers is not in the document]
Comment (ohsh6o) [reply resources/validations/src/ssp.sch:268]: I honestly had not really considered that, ironically. I think we shall put a task down to discuss that with FedRAMP representation.

I was under the impression we could eventually move this to a global variable when we figure out some of the XSpec improvements. I know we only are targeting one `o:description` and `o:remarks`, but that node is contextually different in different places. I am focusing on `o:statement/o:by-component/o:description` and `o:statement/o:by-component/o:remarks`, 20 characters seem like a logical starting point for our previous conversation and thought others would be different.

Added 18F/fedramp-automation#37 to not lose track of this.

File: resources/validations/src/ssp.sch [hunk 2]
```diff
@@ -236,5 +236,49 @@
         <sch:assert role="error" id="invalid-implementation-status" test="not(exists($corrections))">Invalid status '<sch:value-of select="$status"/>' for <sch:value-of select="./@control-id"/>, must be <sch:value-of select="$corrections"/></sch:assert>
     </sch:rule>
 
+    <sch:rule context="/o:system-security-plan/o:control-implementation/o:implemented-requirement/o:statement">
+        <sch:let name="required-components-count" value="1"/>
+        <sch:let name="required-length" value="20"/>
```
[the remark this answers is not in the document]
Comment (ohsh6o) [reply resources/validations/src/ssp.sch:241]: Per conversation with Brian and Wendell. and most importantly the [Guide to FedRAMP OSCAL-formatted SSPs](https://github.com/GSA/fedramp-automation/blob/c56b4b8b02f2e8ed4ecd21ed450771f19d45b14a/documents/Guide_to_OSCAL-based_FedRAMP_System_Security_Plans_(SSP).pdf), we have implemented context in a way that the rules will not fire if `o:by-component` is missing, so we do not check the length unless we have to. But, we still want an error in the event we have a `o:statement` without the required `o:by-component`. The spec document says that is now required.

**Pull request merged**